### PR TITLE
[SPARK-33354][FOLLOWUP][DOC] Shorten the table width of ANSI compliance casting document

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

This is a follow-up of https://github.com/apache/spark/pull/30260
It shortens the table width of ANSI compliance casting document.

### Why are the changes needed?

The table is too wide and the UI of doc site is broken if we scroll the page to right side.
Screen Shot 2021-01-14 at 3 04 57 PM

### Does this PR introduce _any_ user-facing change?

Minor document change

### How was this patch tested?

Build doc site locally and preview:
Screen Shot 2021-01-14 at 4 44 30 PM

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @maropu @HyukjinKwon 
It's minor but I want to make sure the ANSI doc looks fine

**Review by HyukjinKwon — APPROVED** [review]

File: docs/sql-ref-ansi-compliance.md [hunk 1]
```diff
@@ -69,19 +69,19 @@ The type conversion of Spark ANSI mode follows the syntax rules of section 6.13
 
  The valid combinations of target data type and source data type in a `CAST` expression are given by the following table.
 “Y” indicates that the combination is syntactically valid without restriction and “N” indicates that the combination is not valid.
-    
-| From\To   | NumericType | StringType | DateType | TimestampType | IntervalType | BooleanType | BinaryType | ArrayType | MapType | StructType |
+
+| Source\Target | Numeric | String | Date | Timestamp | Interval | Boolean | Binary | Array | Map | Struct |
```
Comment (HyukjinKwon) [inline docs/sql-ref-ansi-compliance.md:73]: We could even make it lower-cased and say these are simple strings or catalog strings .. but I think it's fine either way.

Comment (gengliangwang) [reply docs/sql-ref-ansi-compliance.md:73]: Considering there are "Y"s and "N"s in the table, let's keep the current way.

Comment (HyukjinKwon) [reply docs/sql-ref-ansi-compliance.md:73]: 👌

**Review by dongjoon-hyun — APPROVED** [review]

**Pull request closed without merging**